VirtoCommerce.Storefront/Controllers/Api/ApiOrderController.cs

```diff
@@ -180,6 +180,28 @@ public async Task<ActionResult> GetInvoicePdf(string orderNumber)
             return File(stream, "application/pdf");
         }
 
+        // PUT: storefrontapi/orders/{orderNumber}/status
+        [HttpPut("{orderNumber}/status")]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult> ChangeOrderStatus(string orderNumber, [FromBody] dynamic payload)
+        {
+            var newStatus = payload.newStatus.ToString();
```

Comment:
How do you know that `dynamic payload` has `newStatus` field?

Reply:
Definitely you should to avoid of using the `dynamic` type  as type for formal  parameters for API methods,  since this type doesn't have a definition in  the `OpenApi`  standard

Reply:
Actually, I've added dynamic type here only to simplify to limit the diffs and to localize all changes in one method and do not add extra classes and so on. If you think that adding extra request models it's ok in this case then we certainly will add them)
Also, we know for sure that payload will contain this field because we build the payload on the client-side.